classes/fields/avatar.php

```diff
@@ -79,39 +79,161 @@ public function input( $name, $value = null, $options = null, $pod = null, $id =
 	 */
 	public function get_avatar( $avatar, $id_or_email, $size, $default = '', $alt = '' ) {
 
-		// Don't replace for the Avatars section of the Discussion settings page.
-		if ( is_admin() ) {
-			$current_screen = get_current_screen();
+		if ( ! $this->allow_avatar_overwrite() ) {
+			return $avatar;
+		}
+
+		$user_id = $this->get_avatar_user_id( $id_or_email );
+
+		if ( 0 < $user_id && ! empty( PodsMeta::$user ) ) {
+			$avatar_cached = pods_cache_get( $user_id . '-' . $size, 'pods_avatars' );
+
+			if ( ! empty( $avatar_cached ) ) {
+				$avatar = $avatar_cached;
+			} else {
+
+				$user_avatar = $this->get_avatar_id( $user_id );
+
+				if ( ! empty( $user_avatar ) ) {
+					$attributes = array(
```

Comment:
:no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: b7e81f8103e53e39d008e5ba911cc27b</sub>